Classes/Notifications/NotificationsViewController.swift

```diff
@@ -20,24 +20,41 @@ TabNavRootViewControllerType,
 BaseListViewControllerDataSource,
 FlatCacheListener {
 
+    enum InboxType {
+        case unread
+        case repo(owner: String, name: String)
+        case all
+    }
+
     private let client: NotificationClient
     private let foreground = ForegroundHandler(threshold: 5 * 60)
-    private let showRead: Bool
+    private let inboxType: InboxType
     private var notificationIDs = [String]()
+    private var subscriptionController: NotificationSubscriptionsController?
 
     // set to nil and update to dismiss the rating control
     private var ratingToken: RatingToken? = RatingController.inFeedToken()
 
-    init(client: NotificationClient, showRead: Bool) {
+    init(client: NotificationClient, inboxType: InboxType) {
         self.client = client
-        self.showRead = showRead
+        self.inboxType = inboxType
 
         super.init(
             emptyErrorMessage: NSLocalizedString("Cannot load your inbox.", comment: ""),
             dataSource: self
         )
 
         self.foreground.delegate = self
+
+        switch inboxType {
```

Comment:
What about doing this logic in an `extension NotificationsViewController.InboxType` to prevent cluttering init?